src/utils.h

```diff
@@ -35,41 +38,53 @@ extern log_level_t log_level;
 extern char *log_cid;
 extern gboolean use_syslog;
 
+static inline const char *get_errno_string(void)
+{
+	static __thread char errbuf[256];
+	int saved_errno = errno;
+	snprintf(errbuf, sizeof(errbuf), "%s", strerror(saved_errno));
```

Comment:
`if … >= sizeof(errbuf) { abort() }` or something? (`sprintf("[unexpected error %d"])` is an option but unreasonable)